packages/core/src/layout/HomepageTimer/HomepageTimer.tsx

```diff
@@ -14,59 +14,85 @@
  * limitations under the License.
  */
 
-import React, { FC } from 'react';
+import React from 'react';
 import { HeaderLabel } from '../HeaderLabel';
+import { ConfigApi, useApi, configApiRef } from '@backstage/core-api';
 
 const timeFormat = { hour: '2-digit', minute: '2-digit' };
-const utcOptions = { timeZone: 'UTC', ...timeFormat };
-const nycOptions = { timeZone: 'America/New_York', ...timeFormat };
-const tyoOptions = { timeZone: 'Asia/Tokyo', ...timeFormat };
-const stoOptions = { timeZone: 'Europe/Stockholm', ...timeFormat };
-
-const defaultTimes = {
-  timeNY: '',
-  timeUTC: '',
-  timeTYO: '',
-  timeSTO: '',
-};
 
-function getTimes() {
+class TimeObj {
+  time: string;
+  label: string;
+
+  constructor(time: string, label: string) {
+    this.time = time;
+    this.label = label;
+  }
+}
+
+function getTimes(configApi: ConfigApi) {
   const d = new Date();
   const lang = window.navigator.language;
 
-  // Using the browser native toLocaleTimeString instead of huge moment-tz
-  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString
-  const timeNY = d.toLocaleTimeString(lang, nycOptions);
-  const timeUTC = d.toLocaleTimeString(lang, utcOptions);
-  const timeTYO = d.toLocaleTimeString(lang, tyoOptions);
-  const timeSTO = d.toLocaleTimeString(lang, stoOptions);
+  const clocks: TimeObj[] = [];
+
+  if (!configApi.has('homepage.clocks')) {
+    return clocks;
+  }
+
+  const clockConfigs = configApi.getOptionalConfigArray('homepage.clocks');
 
-  return { timeNY, timeUTC, timeTYO, timeSTO };
+  if (!clockConfigs) {
```

Comment:
the two checks here at line 45 and line 39 seem extra verbose to me and feels like there would be a more idiomatic way to do it in Typescript.

Reply:
Since you already checked for the existence you can just do `configApi.getConfigArray` and return type won't have `| undefined`